app.rb

```diff
@@ -0,0 +1,129 @@
+require_relative 'classes/book'
+require_relative 'classes/student'
+require_relative 'classes/teacher'
+require_relative 'classes/rental'
+
+class App
+  def initialize()
+    puts
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def take_input_with_label(label)
+    print "#{label}: "
+    gets.chomp
+  end
+
+  def person_get_option
+    puts 'What you want to create:'
+    puts "\nPress 1 to add - A teacher Or Press 2 to add - A student"
+    puts 'Press any other key to return to main menu'
+
+    input = gets.chomp
+    return "\n#{input} is not a valid input to create a person" unless (1..2).include?(input.to_i)
+
+    input.to_i
+  end
+
+  def list_all_books
+    @books.each.with_index(1) do |book, index|
+      puts "\n#{index}.) Book's Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people(indexed)
+    text = ''
+
+    @people.each.with_index(1) do |person, i|
+      type = person.instance_of?(Student) ? 'Student' : 'Teacher'
+      data = "[#{type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+
+      text += "\n#{i}) #{data}" if indexed
+      text += "\n#{data}" unless indexed
+    end
+
+    text
+  end
```

Comment:
- [ ] Hello and nice work on adding the UI but this definition of `list_all_people` is not rendering the added students or teachers, so please fix it **hint: you've provided a wrong argument**.

![Capture d’écran 2022-08-27 132841](https://user-images.githubusercontent.com/61562626/187030429-c2090e4b-780b-4fb1-93e6-01a40a26b3f3.png)